src/wrapper-utils.js

```diff
@@ -0,0 +1,191 @@
+/* global cloneInto, exportFunction, mozProxies */
+// Tests don't define this variable so fallback to behave like chrome
+const hasMozProxies = typeof mozProxies !== 'undefined' ? mozProxies : false
+const globalObj = typeof window === 'undefined' ? globalThis : window
+const getOwnPropertyDescriptor = Object.getOwnPropertyDescriptor
+const functionToString = Function.prototype.toString
+const objectKeys = Object.keys
+const NativeProxy = Proxy
+
+export function defineProperty (object, propertyName, descriptor) {
+    if (hasMozProxies) {
+        const usedObj = object.wrappedJSObject
+        const UsedObjectInterface = globalObj.wrappedJSObject.Object
+        const definedDescriptor = new UsedObjectInterface();
+        ['configurable', 'enumerable', 'value', 'writable'].forEach((propertyName) => {
+            if (propertyName in descriptor) {
+                definedDescriptor[propertyName] = cloneInto(
+                    descriptor[propertyName],
+                    definedDescriptor,
+                    { cloneFunctions: true })
+            }
+        });
+        ['get', 'set'].forEach((methodName) => {
+            if (methodName in descriptor) {
+                exportFunction(descriptor[methodName], definedDescriptor, { defineAs: methodName })
+            }
+        })
+        UsedObjectInterface.defineProperty(usedObj, propertyName, definedDescriptor)
+    } else {
+        Object.defineProperty(object, propertyName, descriptor)
+    }
+}
+
+/**
+ * add a fake toString() method to a wrapper function to resemble the original function
+ * @param {*} newFn
+ * @param {*} origFn
+ */
+function wrapToString (newFn, origFn) {
+    if (!newFn || !origFn) {
+        return
+    }
+    newFn.toString = function () {
+        if (this === newFn) {
+            return functionToString.call(origFn)
+        } else {
+            return functionToString.call(this)
+        }
+    }
+}
+
+function getObjectFromPath (objPath, base = globalObj) {
```

Comment:
I think something like this might already exist, perhaps in runtimeChecks? (I feel like I've wrote this 50 times in the last month, so maybe we should move it over if there is one)

Reply:
The closest I could find is https://github.com/duckduckgo/content-scope-scripts/blob/main/src/features/runtime-checks/script-overload.js#L216, but it's not quite the same. Is there another function I can reuse?

Reply:
Yeah maybe not then, ignore :)